src/main/kotlin/de/itemis/mps/gradle/generate/Plugin.kt

```diff
@@ -28,48 +27,25 @@ open class GenerateMpsProjectPlugin : Plugin<Project> {
             val mpsLocation = extension.mpsLocation ?: File(project.buildDir, "mps")
 
             afterEvaluate {
-                val mpsVersion = extension
-                        .mpsConfig
-                        .resolvedConfiguration
-                        .firstLevelModuleDependencies.find { it.moduleGroup == "com.jetbrains" && it.moduleName == "mps" }
-                        ?.moduleVersion ?: throw GradleException("MPS configuration doesn't contain MPS")
+                val mpsVersion = extension.getMPSVersion()
```

Comment:
What do you think of removing `afterEvaluate` and executing the logic only in case it would be needed in the relevant task?

For instance, `./gradlew help` does not need to compute MPS version.

Reply:
Nice catch. I'm gonna create a separate PR that removes `afterevalutate` for all other plugins as well. Then we ship this change as a whole in 1.5.